chore(*) remove Kubernetes submodule

jpeach opened the pull request:
### Summary

Remove the Kubernetes Go submodule.

### Full changelog

N/A

### Issues resolved

Updates #2200 

### Documentation

N/A

### Testing

- [x] Unit tests
- [x] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes 

### Backwards compatibility

- [x] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

lobkovilya commented:
I think a good amount of Makefile targets also could be deleted (probably all of them), as well as `Dockerfile` and `PROJECT`

jpeach commented:
> I think a good amount of Makefile targets also could be deleted (probably all of them), as well as `Dockerfile` and `PROJECT`

I removed these and cleaned up the Makefile a bit/

jpeach commented on pkg/plugins/resources/k8s/native/config/rbac/role.yaml:
```diff
@@ -66,14 +66,6 @@ rules:
-  verbs:
-  - get
-  - patch
-  - update
```
ProxyTemplate doesn't have a Status field, and it doesn't make sense for it to have a status subresource. So this change is correct, but not sure how it got in there in the first place.

lobkovilya commented:
@jpeach did you check if CI is just flaky or it's something connected to the changes?

jpeach commented:
> @jpeach did you check if CI is just flaky or it's something connected to the changes?

Yeh some  sort of flake. I was able to re-run it to make it clear while I was failing to try to set it up and reproduce locally.
